.github/workflows/release-commit.yml

```diff
@@ -0,0 +1,59 @@
+# Create a new commit
+# This action bumps the Cargo.toml version and updates related files.
+# Updates will affect the following files:
+# - ./Cargo.toml
+# - ./Cargo.lock
+# - ./THIRD-PARTY-LICENSES.md
+# - ./CHANGELOG.md
+name: "Release Commit"
+on:
+  workflow_dispatch:
+    inputs:
+      bump_level:
+        description: "What type of release is this?"
+        required: true
+        type: choice
+        options:
+          - minor
+          - patch
+permissions:
+  contents: write
+
+jobs:
+  setup-environment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Add git tag to release
+        run: ./scripts/create_release_changes ${{ github.event.inputs.text_input }}
+
+      - name: Get new cargo version
+        id: cargo_version
+        run: |
+          VERSION=$(cargo read-manifest | jq -r ".version")
+          echo "value=v$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Ensure this release doesn't already exist
+        run: |
+          if git log --grep="Release ${{ steps.cargo_version.outputs.value }}" --quiet; then
```

Comment:
Is this still necessary if we're checking for an existing tag?

Reply:
I think this will cause a false-positive in the event the release got reverted for whatever reason.

If we're trying to determine is whether HEAD is already released, perhaps this will do:

```sh
git tag --contains <SHA1>
```

Reply:
I'll remove from here because the tag is already checked in the script file. The tags are important though since they are what the changelogs are linking to. The release action tags the commit before it does a binary build. It is possible the binary build fails in some strange case, then we have to manually delete the tag, fix the issue and do a new release. Not necessarily the best way of doing it but it's how it is set up atm. Usually the build doesn't fail if the tests pass though.